### PR TITLE
Use Firebase v1 and Upgrade to v1.0.0

### LINKED ISSUE #119: Drop Python 2 support
What do you think about dropping support of Python 2 in one of future major releases?

Also it would be nice add tests only for Python3 ( in scope of #79) afterwards. It seems like clean and reliable way to go through.

// I actively use this package, so I can help with implementation.

### LINKED ISSUE #122: Feature request: provide a callback with list of deactivated results to backup unsent notifications
I am currently using the library to push notifications and it works very well. However, I need to store unsent notifications in my database in case the user disconnects (in which case I typically delete the device to keep things clean, since one user can have multiple devices). Unsent notifications are:
-those where the user currently has no active device 
-and more importantly where the result has an error, as marked in this function:

```
    def _deactivate_devices_with_error_results(self, registration_ids, results):
        for (index, item) in enumerate(results):
            if 'error' in item:
                error_list = ['MissingRegistration', 'MismatchSenderId', 'InvalidRegistration', 'NotRegistered']
                if item['error'] in error_list:
                    registration_id = registration_ids[index]
                    self.filter(registration_id=registration_id).update(
                        active=False
                    )
                    self._delete_inactive_devices_if_requested(registration_id)
```

I'm interested in a neat way to get a list of all FCMDevice objects who have been updated to `active=False` to be able to save a Notification object (for which I created a Model) in my database. 
My model comprises this field: `recipients = models.ManyToManyField(User, db_index=True, related_name="unread_notifications", blank=True)`
So when the client reconnects, he does a simple GET request to query `user.unread_notifications.all()`, and then I can delete the user from recipients.

### LINKED ISSUE #144: Info: support for FCM HTTP v1 API
Hello,
thank you for your work! I have a question: does the current release use the FCM HTTP v1 API?

### LINKED ISSUE #148: Sending topic messages
Hi @xtrinch,
I am working on using this library to send messages to all users subscribed to a particular topic. The documentation on the same barely covers anything and I cannot find any solution anywhere else on the internet for the same. 

Is it an active feature in the library? If so, can you provide a proper documentation on how to subscribe users to a topic and send topical notifications?

### PR DESCRIPTION
**Pull request by Andrew-Chen-Wang**
Fixes #148 
Fixes #144
Fixes #122
Fixes #119 

Also adds topic subscription handling

This needs a little bit of testing (I didn't test this at all...)

Note: I took out fcm.py since people can just use `firebase_admin.messaging import send, send_all` whenever they need. There's no need to do it here since we have no API key to use. I've made the queryset object for deactivating public, however, if people wanted to maintain compatibility. If requested, I can reimplement fcm.py as an API file.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,17 +33,14 @@
         "fcm_django/management",
```
Comment (xtrinch) [inline setup.py:33]: needs to be removed

File: setup.py [hunk 2]
```diff
@@ -32,17 +33,14 @@
...
         "fcm_django/management/commands",
```
Comment (xtrinch) [inline setup.py:34]: needs to be removed

File: fcm_django/models.py [hunk 3]
```diff
@@ -41,222 +45,301 @@ def get_queryset(self):
...
-
-    @staticmethod
-    def _delete_inactive_device_if_requested(device):
-        if SETTINGS["DELETE_INACTIVE_DEVICES"]:
-            device.delete()
+    @classmethod
+    def deactivate_devices_with_error_result(
+        cls, registration_id, firebase_exc, name=None
+    ) -> List[str]:
+        return cls.objects.deactivate_devices_with_error_results(
```
Comment (xtrinch) [inline fcm_django/models.py:337]: this line is throwing an error,
image

**Comment by xtrinch** [comment]
I've created a branch on the demo project using this PR to facilitate the testing:
https://github.com/xtrinch/fcm-django-web-demo/tree/feature/fcm-django-1.0.0

File: fcm_django/admin.py [hunk 4]
```diff
@@ -1,79 +1,104 @@
...
-            total_failure += failure
-            errors.append(str(response))
+                )
+                if type(response) != SendResponse:
+                    total_failure += 1
 
-            if bulk:
-                break
-
-        if ret:
```
Comment (xtrinch) [inline fcm_django/admin.py:63]: This should not be removed, as it showed the actual FCM errors to the user. Showing merely "failed to send" is fairly useless

**Comment by xtrinch** [comment]
Other than that I seem to be able to send notifications just fine, so looking promising!

**Comment by daadu** [comment]
@Andrew-Chen-Wang Thanks for working on this.

File: fcm_django/admin.py (hunk 4 above)
Comment (daadu) [reply fcm_django/admin.py:63]: +1

**Review by daadu — COMMENTED** [review]
I have reviewed the code, mostly looks good to me. I have few queries/suggestions:
1. Looks like for migration, no changes need to be done (to code or data)? Am I right? If not then we should write migration guide (for <v1 to v1 users). except "google credential file" instead of key in settings?

I haven't tested the code yet, this is just from reading the code.

**Comment by Andrew-Chen-Wang** [comment]
@xtrinch  https://github.com/xtrinch/fcm-django-web-demo/pull/23 adjusted demo readme. Thanks for creating that new branch!

File: README.rst [hunk 5]
```diff
@@ -25,39 +27,105 @@ Functionality:
  - Django rest framework viewsets
 
 Demo javascript client project
--------------------
+------------------------------
+Note: the current demo project uses fcm-django < v1
```
Comment (xtrinch) [inline README.rst:31]: We can omit that since we'll be updating the demo project too

File: requirements.txt [hunk 6]
```diff
@@ -1 +1,7 @@
-pyfcm==1.4.9
+Django
```
Comment (xtrinch) [inline requirements.txt:1]: since we only support django 2.2+, would probably be good to specify this here

File: requirements.txt [hunk 7]
```diff
@@ -1 +1,7 @@
...
+django-tastypie
```
Comment (xtrinch) [inline requirements.txt:2]: It would be good to specify all minimum versions of the required packages

**Comment by xtrinch** [comment]
A couple of merge conflicts arose from one of your other PRs. Other than that it looks good to me, thanks! Will merge after that and create the 1.0.0 release

**Comment by Andrew-Chen-Wang** [comment]
@xtrinch I highly recommend you first create a release with current master so that the v0 users get #159. Then merge this for v1

**Comment by xtrinch** [comment]
Alright. I'll also create an archive branch for the v0 so we don't lose it.

**Review by xtrinch — APPROVED** [review]

**Comment by xtrinch** [comment]
Released last pyfcm version under 0.3.11 and the 1.0.0 is now live :rocket: Thanks to @Andrew-Chen-Wang

**Comment by Andrew-Chen-Wang** [comment]
thanks for the quick responses and testing @xtrinch !

**Pull request merged**